align/compiler/create_array_hierarchy.py

```diff
@@ -219,14 +210,26 @@ def add_align_block_const(self):
             h_blocks = [inst for inst in inst_list
                         if inst in self.graph]
             if len(h_blocks) > 0:
+                for const in self.iconst:
+                    if isinstance(const, constraint.Align):
+                        if set(const.instances).issubset(set(h_blocks)):
+                            return  # duplicate constraint
+                # TODO: try/except for auto-generated constraints
                 with set_context(self.iconst):
-                    self.iconst.append(constraint.Align(line="h_center", instances=h_blocks))
-            # del self.match_pairs[key]
+                    self.iconst.append(constraint.SameTemplate(instances=h_blocks))
+                    # If the connectivity is identical for all blocks, ordering does not matter
```

Comment:
@kkunal1408 , please review.

Reply:
I did not get this check.
e.g., why two options. How do these two scenarios differ in terms of array placement?

array1
```
x0 a b name
x1 a c name
x2 a d name
``` 
and array2
```
x0 a b name
x1 a b name
x2 a b name
``` 
I guess both can be handled using align constraint and hierarchy creation is not required. Am I missing something?

Reply:
array1 gets Align, array2 gets AlignInOrder as ordering does not matter.